kv/tikv/server.go

```diff
@@ -36,9 +36,46 @@ type InnerServer interface {
 	Snapshot(stream tikvpb.Tikv_SnapshotServer) error
 }
 
-func NewServer(innerServer InnerServer) *Server {
+// Scheduler takes Commands and runs them asynchronously. It is up to implementations to decide the scheduling policy.
+type Scheduler interface {
+	Run(Command) <-chan RespResult
+	Stop()
```

Comment:
how about moving them to a new file, so we can easily clean them later?

Reply:
Sure, but do you mind if I do it in the next PR? I've already built on this stuff so rebasing will be a bit tricky

Reply:
okay